### PR TITLE
Added functionality for generating fake idq data

### PR DESCRIPTION
**Pull request by maxtrevor**
add_state_dq_vectors.py now has options for adding fake idq data to frames

### CODE REVIEW

**Comment by maxtrevor** [comment]
@titodalcanton

File: add_state_dq_vectors.py [hunk 1]
```diff
@@ -91,6 +95,20 @@
     out_channel_names.append(args.dq_vector)
     out_timeseries.append(dq_ts)
 
+# add iDQ data
+
+if args.idq_channel is not None:
+    #generate a fake idq timeseries
+    idq_dt 1. / 128.
```
Comment (titodalcanton) [inline add_state_dq_vectors.py]: Looks like you have a typo here, missing `=`?

File: add_state_dq_vectors.py [hunk 2]
```diff
@@ -91,6 +95,20 @@
...
+    idq_size = int(strain.duration / idq_dt)
+    rng = default_rng(123)
```
Comment (titodalcanton) [inline add_state_dq_vectors.py]: I suggest taking the random seed from an argparse argument.

**Comment by titodalcanton** [comment]
Thanks @maxtrevor. Apart from the minor points above, I just wanted to double check that a random normal timeseries is the most useful simulation here. Would you not rather simulate a quantity that has some value most of the time, and a few occasional peaks at times you specify from the command line? Basically like the existin `--dq-bad*` options do?

**Comment by maxtrevor** [comment]
> Thanks @maxtrevor. Apart from the minor points above, I just wanted to double check that a random normal timeseries is the most useful simulation here. Would you not rather simulate a quantity that has some value most of the time, and a few occasional peaks at times you specify from the command line? Basically like the existin `--dq-bad*` options do?

Hmm I guess that would be useful for the future. The difference of idq vs the state and dq vectors is that idq does not take on discrete values, though in practice it is certainly not a normal distribution either. Maybe a good way to simulate it would be to create a random normal timeseries and then add some constant to the data at and around some times specified from the command line?

**Comment by titodalcanton** [comment]
Right, I did not mean for it to take literally only two values, but to dwell around one value most of the time, and jump close to another value during discrete and short periods (i.e. the moments where you want your simulated glitches to be). Basically like this: https://en.wikipedia.org/wiki/Burst_noise#/media/File:Popcorn_noise_graph.png

**Comment by maxtrevor** [comment]
Hi Tito,
I've had a lot of difficulties with getting this to correctly add data to strain gwfs generated by pycbc_condition_strain . I keep getting issues relating to the start and end times of TimeSeries needing to be integers, and needing to match between TimeSeries in the same frame.
My solution has been to make a new executable in pycbc to handle generating fake frames including both fake strain and other channels and writing them all to a frame file simultaneously. I've created a pull request to merge that executable into the main pycbc repo: https://github.com/gwastro/pycbc/pull/3992

**Comment by titodalcanton** [comment]
@maxtrevor do we still need this?

**Comment by maxtrevor** [comment]
We don't need this

**Pull request closed without merging**